db/users.go

```diff
@@ -83,6 +83,28 @@ func (dbc *DatabaseContext) GetPrincipal(name string, isUser bool) (info *Princi
 	return
 }
 
+func (db *DatabaseContext) DeletePrincipal(name string, isUser bool, purge bool) (found bool, err error) {
```

Comment:
There doesn't really seem to be that much common code between the delete handling for role and user - I think it would make the code much cleaner and easier to maintain if we just switch to DeleteUser and DeleteRole down in auth, and only need this intermediary in the delete role case (to grab the seq).

Reply:
Moved user back to basically using same as before and split the user and role delete.